pkg/operator/ceph/object/objectstore.go

```diff
@@ -392,55 +392,46 @@ func getZoneEndpoints(objContext *Context, serviceEndpoint string) ([]string, bo
 	return zoneEndpointsList, isEndpointAlreadyExists, nil
 }
 
+func createMultisiteConfigurations(objContext *Context, configType, configTypeArg string, args ...string) error {
+	// get the multisite config before creating
+	output, getconfigErr := RunAdminCommandNoMultisite(objContext, true, configType, "get", configTypeArg)
+	if getconfigErr != nil {
+		// ENOENT means "No such file or directory"
+		if code, err := exec.ExtractExitCode(getconfigErr); err == nil && code == int(syscall.ENOENT) {
+			// create the object if it doesn't exist yet
+			output, err := RunAdminCommandNoMultisite(objContext, false, args...)
+			if err != nil {
+				return errorOrIsNotFound(err, "failed to create ceph %q %q, for reason %q", configType, configTypeArg, output)
+			}
+			logger.Debugf("created %q %q", configType, configTypeArg)
+		} else {
+			return errorOrIsNotFound(getconfigErr, "'radosgw-admin %q get' failed with code %d, for reason %q", configType, strconv.Itoa(code), output)
+		}
+	}
+
+	return nil
+}
+
 func createMultisite(objContext *Context, endpointArg string) error {
 	logger.Debugf("creating realm, zone group, zone for object-store %v", objContext.Name)
 
 	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
 	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", objContext.Zone)
 
```

Comment:
IMHO better to set these Args in createMultisiteConfigurations(), pass `realm`, `zone`, `zonegroup` as configType then populate other values based on switch case

Reply:
If we need to call all the functions 3 times for all the cases so we cannot have switch, over them, 

Or we need to create an array of 3 values and call it on switch, SO I think we calling directly would be fine,right?

Reply:
Sounds good to me